cmd/common/registry.go

```diff
@@ -419,7 +455,37 @@ func buildCommandFromGadget(
 				if len(hiddenColumnTags) > 0 {
 					hiddenTags = append(hiddenTags, hiddenColumnTags...)
 				}
-				requestedColumns = parser.GetDefaultColumns(hiddenTags...)
+				requestedColumns = append(requestedColumns, parser.GetDefaultColumns(hiddenTags...)...)
+			}
+
+			// Add/remove relative column requests
+			if len(addCols) > 0 || len(removeCols) > 0 {
+				newRequestedColumns := make([]string, 0)
+				for _, col := range requestedColumns {
+					if containsColumn(removeCols, col) {
+						continue
+					}
+					newRequestedColumns = append(newRequestedColumns, col)
+				}
+				// add remaining columns
+				for _, col := range addCols {
+					if containsColumn(newRequestedColumns, col) || containsColumn(removeCols, col) {
+						continue
+					}
+					newRequestedColumns = append(newRequestedColumns, col)
+				}
+				requestedColumns = newRequestedColumns
+			}
```

Comment:
> (edit: order can be challenging to handle)

Yep, I switched to slice because of ordering. It can be annoying if columns are changing order on each run!